src/Filter/RoleFilter.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Everlution\NavigationBundle\Filter;
+
+use Everlution\Navigation\Filter\FilterInterface;
+use Everlution\Navigation\Item;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+
+/**
+ * @author Klinec <matus.pavlik@everlution.com>
+ */
+class RoleFilter implements FilterInterface
+{
+    /**
+     * @var AuthorizationCheckerInterface
+     */
+    private $authChecker;
+
+    /**
+     * @param AuthorizationCheckerInterface $authChecker
+     */
+    public function __construct(AuthorizationCheckerInterface $authChecker)
```

Comment:
I really don't like short variable names - your IDE will autocomplete it for you, don't worry about it. Please just use whole `$authorizationChecker` name